packages/markdown/processor/parse/magic-block-parser.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable consistent-return */
+const RGXP = /^\[block:(.*)\]([^]+?)\[\/block\]/;
+
+function tokenize(eat, value) {
+  let [match, type, json] = RGXP.exec(value) || [];
+
+  if (!type) return;
+
+  match = match.trim();
+  type = type.trim();
+  json = (json && JSON.parse(json)) || {};
+
+  switch (type) {
+    case 'code':
+      return eat(match)({
+        type: 'rdme-wrap',
+        className: 'tabs',
+        data: { hName: 'rdme-wrap' },
+        children: json.codes.map(obj => ({
+          type: 'code',
+          value: obj.code,
+          meta: obj.name || null,
+          lang: obj.language,
+          className: 'tab-panel',
+        })),
+      });
+    case 'api-header':
+      return eat(match)({
+        type: 'heading',
+        depth: json.level || 1,
+        children: this.tokenizeInline(json.title, eat.now()),
+      });
+    case 'image':
+      return eat(match)(json.images.map(img => {
+        const [url, alt] = img.image;
+        return {
+          type: 'image',
+          title: img.caption, // this.tokenizeInline(img.caption, eat.now()),
+          url,
+          alt,
+        };
+      })[0]);
+    case 'callout': {
+      json.type = {
```

Comment:
Can this be reused in some way? https://github.com/readmeio/api-explorer/pull/306/files#diff-01b7224cc1d3e9f2a05bd48228c1bda9R12-R21

Reply:
That link isn't working for me; do you remember which file you were referencing specifically?